pkg/broker/filter/filter_handler.go

```diff
@@ -182,8 +182,8 @@ func (r *Handler) serveHTTP(ctx context.Context, event cloudevents.Event, resp *
 		// framework returns a 500 to the caller, so the Channel would send this repeatedly.
 		return nil
 	}
-	delete(originalV3.Extensions, ttlKey)
-	event.Context = originalV3
+	delete(originalV1.Extensions, ttlKey)
```

Comment:
How about adding a Delete method to Extensions and handle this correctly in cloudEvents sdk?